### PR TITLE
Store credentials sign up in database

### LINKED ISSUE #30: Store the inputted credentials from the sign up custom form in the database
[Support custom profile (name, profile picture, favorite music type ...)](https://github.com/OrkestApp/sdp_2023/issues/16)

### PR DESCRIPTION
**Pull request by yasmineben510**
Updated the viewModel associated to the Sign Up activity such that the infos related to the newly created user are stored in the database.

It checks if a username already exists in the database and displays an error, otherwise, it send the user to the main activity

Finally, to test the UI, I created a mockAuthView that simulates a database returning false for a used username and true for a valid one.

The ViewModel was tested using the firestore emulator which was also integrated to the CI. The emulator doesn't automatically resets its values between each test, so we must make sure to clear it in our tests if desired.

This PR closes #30

### CODE REVIEW

File: app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt [hunk 1]
```diff
@@ -11,51 +13,72 @@ import androidx.compose.runtime.remember
...
-                val expanded = remember { mutableStateOf(false) }
-
-
+                //Title of the page
                 Text(
                     text = "Create Your Profile",
                     style = TextStyle(fontSize = 40.sp, fontFamily = FontFamily.Cursive)
                 )
 
                 Spacer(modifier = Modifier.height(15.dp))
```
Comment (RoCKnD79) [inline app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt:65]: I don't know what's the best practice and it's already well readable as it it, but maybe a good idea would be to modularize the blocks of code that represent different elements.

In this case you could call a function called "Title()" that contains the given block of code

File: app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt [hunk 2]
```diff
@@ -11,51 +13,72 @@ import androidx.compose.runtime.remember
...
```
Comment (RoCKnD79) [inline app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt:64]: You repeat this same line a few times in the code. Unless you intend on modifying the space between elements frequently I think you could store this into a variable.

File: app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt [hunk 3]
```diff
@@ -65,9 +88,8 @@ fun SignUpForm(navController: NavController, viewModel: AuthViewModel) {
...
 
+                // Box with button to choose the service provider from a dropDownMenu
                 Box {
-
-
                     Button(
```
Comment (RoCKnD79) [inline app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt:93]: Same comment as before, I'd store this block of code in a function that you'd then call here.
I'd say it makes it easier to read and maintain the code

**Review by RoCKnD79 — COMMENTED** [review]
The code in general looks good to me, concise, efficient, I don't have anything to say on the functionality !
Also very nice that you added comments everywhere which makes the code easier to understand :)
The only comment I'd make is on modularization in the signupform where blocks of code could be instead contained in functions, but I'd say it's for you to decide. Perhaps you thought the code is already not as large as it could be and therefore you can leave it as is especially since the comments guide the reader quite well.

File: app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt (hunk 3 above)
Comment (RoCKnD79) [reply app/src/main/java/com/github/orkest/View/auth/SignUpForm.kt:93]: Apparently I am incapable of selecting lines correctly, let me try again...

**Comment by Zermelo-101** [comment]
I beleive the changes you made are good, good code, nice structure and respect of the Pattern. Ready to merge

**Review by Zermelo-101 — APPROVED** [review]
I agree with what Roman said, the last changes you made LGTM, ready to merge
